src/util/command_line_handling.cpp

```diff
@@ -355,6 +355,8 @@ namespace hpx { namespace util
                 threads_str = vm["hpx:threads"].as<std::string>();
                 if ("all" == threads_str)
                 {
+                    default_threads = thread::hardware_concurrency();
+                    batch_threads = env.retrieve_number_of_threads();
```

Comment:
This is a change to the original implementation of `--hpx:threads=all`. Could you give a rationale for why it is needed?

Reply:
'all' always considered the batch environment previously.

Reply:
@sithhell Sure, but not at this spot. Did we miss to properly handle the environment earlier?

Reply:
The reason is because if you look a couple of lines above, the default and batch threads are set to those values, but since "threads=cores" is the new default, they immediately get reset to something else, so then inside the `if == all` clause I reset them back to what they would have been if I hadn't made `cores` the new default. The code is a bit messy and probably needs a rewrite, but that should go into a separate issue I think. (also the command line handling code is called twice and I'm puzzled as to why it needs to be done twice, so that should be looked at at the same time)

Reply:
Thanks for the explanations, makes perfect sense.